docs/case-control-studies.md

```diff
@@ -0,0 +1,150 @@
+There are five steps to undertaking a [case-control study](https://en.wikipedia.org/wiki/Case%E2%80%93control_study) with OpenSAFELY:
```

Comment:
Thanks @iaindillingham - just had a read through the documentation and it looks great. My only suggestion would be to add '(or matched cohort study)' after 'case-control study'.

Reply:
Thanks @johntaz. Done in b2d318f.